images/scripts/make-image-tag.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# Copyright 2017-2020 Authors of Cilium
+# SPDX-License-Identifier: Apache-2.0
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+# This script provides two image tagging mechanisms.
+#
+# For general images that use most of the tree as input, it's most sensible to
+# use git commit hash as a tag.
+#
+# For images that use contents of a subdirectory as input, it's convenient to use
+# a git tree hash. Running `git show` with tree hash based tag will display the
+# contents of the subdirectory that was used as build input, mitigating any doubts
+# in what was used to build this image.
+#
+# For both types of tags To differentiate any non-authoritative builds, i.e.
+# builds from development branches, `-dev` suffix is added. Any builds that may
+# include uncommitted changes will have `-wip` tag.
+
+# TODO: determine when the head is also a release tag; that it can be done with
+# `git name-rev --name-only --tags --no-undefined HEAD` however Cilium usually
+# gets two tags one with `v` prefix and one without, beacause of that the output
+# of `git name-rev` maybe one or the other; to implement that with certainty
+# release process/automation needs to be reviewed in detail
```

Comment:
There's some background here where IIRC readthedocs doesn't recognize `vX.Y.Z` tags so we have to tag `X.Y.Z` to get RTD to generate the "stable" docs based on a newly released version; at the same time IIRC Github encourages `vX.Y.Z` style tags via its releases page mechanisms, I don't recall the details.

One potential mitigation may be moving docs build into our own infrastructure and just pushing static pages to a simpler, dumber webserver. But that's in general a bigger question.

Reply:
(I don't think we ever tried to get RTD to recognize `vX.Y.Z`, maybe that's another path to explore)

Reply:
I actually fixed this TODO, will push new version soon.